.script/tests/KqlvalidationsTests/KqlValidationTests.cs

```diff
@@ -95,6 +97,23 @@ private bool ShouldSkipTemplateValidation(string templateId)
             var col = (pos - curPos + 1);
             return (curlineIndex + 1, col);
         }
+
+        /// <summary>
+        ///Get detection yaml file from Detection or solution analytics rule folder
+        /// </summary>
+        /// <param name="detectionsYamlFileName">detections Yaml File Name</param>
+        /// <returns>detections yaml file path</returns>
+        private string getDetectionsYamlFile(string detectionsYamlFileName)
+        {
+            try
+            {
+                return Directory.GetFiles(DetectionPath[0], detectionsYamlFileName, SearchOption.AllDirectories).Single();
+            }
+            catch
+            {
+                return  Directory.GetFiles(DetectionPath[1], detectionsYamlFileName, SearchOption.AllDirectories).Where(s => s.Contains("Analytic Rules")).Single();
```

Comment:
So if there is a template in detections and a template in solutions that have the same name, only the one from detections will be validated.
I think it is a problem, since this means that currently you can insert errors to the  CiscoUmbrella templates that are in both directories.